tools/perf/lib/benchmark/runner/ib_read.py

```diff
@@ -60,6 +60,8 @@ def __init__(self, benchmark, config, idfile):
                                                           value))
         # pick the settings predefined for the chosen mode
         self.__tool = self.__benchmark.oneseries['tool']
+        if 'mode' not in self.__benchmark.oneseries:
```

Comment:
There should be no changes in this file in this commit. This commit only adds unit tests.